gateway-ha/src/main/java/io/trino/gateway/ha/router/OAuth2GatewayCookieProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import io.airlift.units.Duration;
+import io.trino.gateway.ha.config.HaGatewayConfiguration;
+
+import java.util.List;
+
+public class OAuth2GatewayCookieProvider
+        extends AbstractModule
+{
+    private final List<String> deletePaths;
+    private final Duration ttl;
+
+    @Inject
+    public OAuth2GatewayCookieProvider(HaGatewayConfiguration configuration)
+    {
+        this.deletePaths = configuration.getOauth2GatewayCookieConfiguration().getDeletePaths();
```

Comment:
Is `configuration.getOauth2GatewayCookieConfiguration()` always non-null?

Reply:
it is initialized in HaGatewayConfiguration, but there are no guarantees